src/spdx_matcher/__init__.py

```diff
@@ -198,142 +201,151 @@ def normalize(license_text, remove_sections=REMOVE_FINGERPRINT):
-                    if match < 0.99:
-                        print(
-                            f"Unable to match {match} full_match:{full_match}  exmplar text "
-                            f"to regexp for exception "
-                            f"{json_detail_data['licenseExceptionId']}  "
-                            f"license_data {json.dumps(license_data)} time {execution_time}"
-                        )
-                    else:
-                        print(
-                            f"Success to match and full_match:{full_match} exmplar text to "
-                            f"regexp for exception {json_detail_data['licenseExceptionId']} "
-                            f"license_data {json.dumps(license_data)} time {execution_time}"
-                        )
-            for k in json_detail_data:
-                if k not in exception:
-                    exception[k] = json_detail_data[k]
-            match_cache["exceptions"][exception["licenseExceptionId"]] = {
-                "name": exception["name"],
-                "regexpForMatch": exception["regexpForMatch"],
-                "matchCost": exception["matchCost"]
-                if "matchCost" in exception
-                else 100.0,
-                "text_length": len(exception["licenseExceptionText"]),
-                "matchConfidence": exception["matchConfidence"],
-            }
-
-        cache_file = os.getenv("SPDX_MATCHER_CACHE_FILE", DEFAULT_CACHE_PATH)
-        with open(cache_file, "wt", encoding="utf-8") as cf:
-            cf.write(json.dumps(match_cache, indent=4))
+
+                if match < 0.99:
+                    print(
+                        f"Unable to match {match} full_match:{full_match}  exmplar "
+                        f"text to regexp for license {json_detail_data['licenseId']}  "
+                        f"license_data {json.dumps(license_data)} time {execution_time}"
+                    )
+                else:
+                    print(
+                        f"Success to match and full_match:{full_match} exmplar text "
+                        f"to regexp for license {json_detail_data['licenseId']} "
+                        f"license_data {json.dumps(license_data)} time {execution_time}"
+                    )
+
+        for k in json_detail_data:
+            if k not in license:
+                license[k] = json_detail_data[k]
+        match_cache["licenses"][license["licenseId"]] = {
+            "name": license["name"],
+            "regexpForMatch": license["regexpForMatch"],
+            "matchCost": license["matchCost"] if "matchCost" in license else 100.0,
+            "text_length": len(license["licenseText"]),
+            "matchConfidence": license["matchConfidence"],
+            "isDeprecatedLicenseId": license["isDeprecatedLicenseId"],
+        }
+
+
+def _process_exception(exception, match_cache):
+    if "detailsUrl" in exception:
+        json_detail_data = _fetch_json(exception["detailsUrl"])
```

Comment:
I change this logic, because I can not get any expection data with the original logic
```diff
-        if "reference" in exception:
-            url = exception["reference"].replace(".", base_url, 1)
-            with urllib.request.urlopen(url, timeout=10) as f:
-                json_detail_data = json.loads(f.read().decode("utf-8"))
+        json_detail_data = _fetch_json(exception["detailsUrl"])
```
And I do not think `url = exception["reference"].replace(".", base_url, 1)` make sense